File: algorithms/easy/roman_to_integer/roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        result = 0
        
        for i in range(0, len(s)):
            if s[i] == 'I':
                if i != len(s)-1:
                    if s[i+1] == 'V' or s[i+1] == 'X':
                        result -= 1
                    else:
                        result += 1
                else:
                    result += 1
                    
            elif s[i] == 'V':
                result += 5
            
            elif s[i] == 'X':
                if i != len(s)-1:
                    if s[i+1] == 'L' or s[i+1] == 'C':
                        result -= 10
                    else:
                        result += 10
                else:
                    result += 10
                    
            elif s[i] == 'L':
                result += 50
            
            elif s[i] == 'C':
                if i != len(s)-1:
                    if s[i+1] == 'D' or s[i+1] == 'M':
                        result -= 100
                    else:
                        result += 100
                else:
                    result += 100
            
            elif s[i] == 'D':
                result += 500
                
            elif s[i] == 'M':
                result += 1000
                
        if result > 0 and result < 4000:
            return result
                    
        return 0
            

File: algorithms/easy/roman_to_integer/roman_to_integer.go
package main

import "strings"

func romanToInt(s string) int {
	r := strings.Split(s, "")
	result := 0

	for i := 0; i < len(r); i++ {
		switch r[i] {
		case "I":
			if i != len(r)-1 {
				if r[i+1] == "V" || r[i+1] == "X" {
					result--
				} else {
					result++
				}
			} else {
				result++
			}

		case "V":
			result += 5

		case "X":
			if i != len(r)-1 {
				if r[i+1] == "L" || r[i+1] == "C" {
					result -= 10
				} else {
					result += 10
				}
			} else {
				result += 10
			}

		case "L":
			result += 50

		case "C":
			if i != len(r)-1 {
				if r[i+1] == "D" || r[i+1] == "M" {
					result -= 100
				} else {
					result += 100
				}
			} else {
				result += 100
			}

		case "D":
			result += 500

		case "M":
			result += 1000
		}
	}

	if result >= 1 && result <= 3999 {
		return result
	}
	return 0
}

func main() {}
